File: dotnet-backend/Dtos/AccountDtos.cs
namespace Wellness.Backup.Api.Dtos;

/// <summary>
/// DTOs for the privacy / account-management endpoints (S-03), mirrored from the
/// Spring Boot backend so the exported JSON is identical across both backends.
/// </summary>
/// <remarks>@author Chua Wei Yi Justin, Tiong Zhong Cheng</remarks>
public sealed record AccountUserProfile(
    long Id,
    string Email,
    string DisplayName,
    string Role,
    DateTime CreatedAt);

public sealed record ChatExport(
    long Id,
    string Question,
    string Answer,
    string? SourceSummary,
    string? ModelName,
    DateTime CreatedAt);

public sealed record AccountExport(
    AccountUserProfile Profile,
    IReadOnlyList<WellnessRecordResponse> WellnessRecords,
    IReadOnlyList<RecommendationResponse> Recommendations,
    IReadOnlyList<ChatExport> ChatMessages,
    DateTime ExportedAt);

/// <summary>
/// Optional password re-confirmation body for the permanent delete. Local
/// password accounts must provide the current password; SSO-only accounts have
/// no app password to re-enter, so the authenticated JWT is enough.
/// </summary>
public sealed record DeleteAccountRequest(string? Password);

File: spring-backend/src/main/java/sg/edu/nus/iss/wellness/dto/AccountDtos.java
package sg.edu.nus.iss.wellness.dto;

import java.time.Instant;
import java.util.List;

/**
 * DTOs for the privacy / account-management endpoints (S-03).
 *
 * <p>{@link AccountExport} is the full, portable copy of everything the app
 * stores about a user, returned by {@code GET /api/account/export}.
 *
 * @author Chua Wei Yi Justin, Tiong Zhong Cheng
 */
public final class AccountDtos {
    private AccountDtos() {
    }

    /** Account profile fields safe to hand back to the owner (no password hash). */
    public record UserProfile(
            Long id,
            String email,
            String displayName,
            java.math.BigDecimal heightCm,
            String role,
            Instant createdAt
    ) {
    }

    /** Profile update payload for the authenticated owner. */
    public record ProfileUpdateRequest(
            java.math.BigDecimal heightCm
    ) {
    }

    /** A stored chatbot exchange, flattened for export. */
    public record ChatExport(
            Long id,
            String question,
            String answer,
            String sourceSummary,
            String modelName,
            Instant createdAt
    ) {
    }

    /** Everything stored about a user, aggregated for a self-service data export. */
    public record AccountExport(
            UserProfile profile,
            List<WellnessDtos.WellnessRecordResponse> wellnessRecords,
            List<RecommendationDtos.RecommendationResponse> recommendations,
            List<ChatExport> chatMessages,
            Instant exportedAt
    ) {
    }

    /**
     * Optional password re-confirmation body for the permanent delete. Local
     * password accounts must provide the current password; SSO-only accounts
     * have no app password to re-enter, so the authenticated JWT is enough.
     */
    public record DeleteAccountRequest(
            String password
    ) {
    }
}
